### PR TITLE
Screen reader labels

### PR DESCRIPTION
**Pull request by AyoobAboosalih**
### Related Issues
- https://github.com/wso2/product-is/issues/15270

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [x] Documentation provided. (Add links if there are any)
- [x] Unit tests provided. (Add links if there are any)
- [x] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/recovery-portal/src/main/webapp/password-reset.jsp [hunk 1]
```diff
@@ -163,7 +163,7 @@
                             %>
 
                            <div class="field">
-                                <label>
+                                <label for="reset-password2">
```
Comment (brionmario) [inline apps/recovery-portal/src/main/webapp/password-reset.jsp:166]: Shall we name this `confirm-password`?

**Review by brionmario — DISMISSED** [review]

File: apps/recovery-portal/src/main/webapp/password-reset.jsp (hunk 1 above)
Comment (AyoobAboosalih) [reply apps/recovery-portal/src/main/webapp/password-reset.jsp]: adjusted with b19bce672761ac6cde9022401cc4ac469aa6fcf0

**Review by kavinda1995 — DISMISSED** [review]

File: apps/recovery-portal/src/main/webapp/password-reset.jsp [hunk 2]
```diff
@@ -163,13 +163,13 @@
...
                            <div class="field">
-                                <label for="reset-password2">
+                                <label for="confirm-password">
                                     <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "Confirm.password")%>
                                 </label>
                                 <div class="ui right icon input">
                                     <input
-                                        id="reset-password2"
-                                        name="reset-password2"
+                                        id="confirm-password"
```
Comment (kavinda1995) [inline apps/recovery-portal/src/main/webapp/password-reset.jsp:171]: Won't we need to change line 224 too with this change?

Comment (AyoobAboosalih) [reply apps/recovery-portal/src/main/webapp/password-reset.jsp:171]: adjusted with 497ebab76c90993725dc62f65153ab2f3b247d2e.

manually tested flow, works as expected

**Review by kavinda1995 — APPROVED** [review]

**Pull request merged**